cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java

```diff
@@ -200,25 +201,32 @@ public boolean isShutdown() {
      * Returns the SQLite Version.
      * @return The SQLite version or "Unknown" if the version could not be determined.
      */
-    public String getSQLiteVersion() {
-        try {
-            return this.submit(new SQLQueueCallable<String>() {
-                @Override
-                public String call(SQLDatabase db) throws Exception {
-                    Cursor cursor = db.rawQuery("SELECT sqlite_version() AS sqlite_version", null);
-                    String sqliteVersion = "";
-                    while(cursor.moveToNext()){
-                        sqliteVersion += cursor.getString(0);
+    public synchronized String getSQLiteVersion() {
+
+        if (this.sqliteVersion == null) {
+            try {
+                this.sqliteVersion = this.submit(new SQLQueueCallable<String>() {
+                    @Override
+                    public String call(SQLDatabase db) throws Exception {
+                        Cursor cursor = db.rawQuery("SELECT sqlite_version()", null);
+
+                        String sqliteVersion = "";
+                        while (cursor.moveToNext()) {
+                            sqliteVersion += cursor.getString(0);
+                        }
+                        return sqliteVersion;
                     }
-                    return sqliteVersion;
-                }
-            }).get();
-        } catch (InterruptedException e) {
-            logger.log(Level.WARNING, "Could not determine SQLite version", e);
-        } catch (ExecutionException e) {
-            logger.log(Level.WARNING, "Could not determine SQLite version", e);
+                }).get();
```

Comment:
I think you need to `return` after this to avoid overwriting your value from the db with `"unknown"`. Or set `"unknown"` inside both the catch blocks.